src/draw_target/clipped.rs

```diff
@@ -95,7 +95,7 @@ mod tests {
         let mut display = MockDisplay::new();
 
         let area = Rectangle::new(Point::new(2, 1), Size::new(2, 4));
-        let mut clipped = display.clipped(&area);
+        let mut clipped = (&mut display).clipped(&area);
```

Comment:
Code like this will be probably be pretty common and I think we should make it a bit nicer:
```suggestion
        let mut clipped = display.by_ref().clipped(&area);
```
This could be implemented by adding a `by_ref` method to the `DrawTarget` trait, like in the `std::io::Read` trait https://doc.rust-lang.org/src/std/io/mod.rs.html#841-881.

Reply:
Ditto. Let me make those changes in the next days. Perhaps the key question is what we have to do w.r.t. `Dimension` and `OriginDimension` from above.